x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_store_data_client.ts

```diff
@@ -0,0 +1,143 @@
+      },
+    });
+    const updatedObj = await this.engineClient.update(savedObj.id, 'started');
+
+    return { ...savedObj.attributes, ...updatedObj.attributes };
+  }
+
+  /**
+  * HOST:START
+    curl -H 'Content-Type: application/json' \
+      -X POST \
+      -H 'kbn-xsrf: true' \
+      -H 'elastic-api-version: 2023-10-31' \
+      http:///elastic:changeme@localhost:5601/api/entity_store/engines/host/start
+  */
+  public async start(entityType: EntityType) {
+    const definition = getEntityDefinition(entityType);
+
+    const savedObj = await this.engineClient.get(entityType).then(ensureEngineExists(entityType));
+
+    if (savedObj.attributes.status !== 'stopped') {
+      throw new Error(
+        `Cannot start Entity engine for ${entityType} when current status is: ${savedObj.attributes.status}`
+      );
+    }
+
+    this.options.logger.debug(`Starting entity store for ${entityType}`);
+    await this.options.entityClient.startEntityDefinition(definition);
+
+    const updatedObj = await this.engineClient.update(savedObj.id, 'started');
+    return updatedObj.attributes;
+  }
+
+  public async stop(entityType: EntityType) {
+    const definition = getEntityDefinition(entityType);
+
+    const savedObj = await this.engineClient.get(entityType).then(ensureEngineExists(entityType));
+
+    if (savedObj.attributes.status !== 'started') {
+      throw new Error(
+        `Cannot stop Entity engine for ${entityType} when current status is: ${savedObj.attributes.status}`
+      );
+    }
+
+    this.options.logger.debug(`Stopping entity store for ${entityType}`);
+    await this.options.entityClient.stopEntityDefinition(definition);
+
+    const updatedObj = await this.engineClient.update(savedObj.id, 'stopped');
+    return updatedObj.attributes;
+  }
+
+  public async get(entityType: EntityType) {
+    return this.engineClient.get(entityType).then(ensureEngineExists(entityType));
+  }
+
+  /**
+   * LIST
+    curl -H 'Content-Type: application/json' \
+      -X GET \
+      -H 'kbn-xsrf: true' \
+      -H 'elastic-api-version: 2023-10-31' \
+      http:///elastic:changeme@localhost:5601/api/entity_store/engines
+  */
+  public async list() {
+    return this.options.soClient
+      .find<EngineDescriptor>({
+        type: entityEngineDescriptorTypeName,
+      })
+      .then(({ saved_objects: engines }) => ({
+        engines: engines.map((engine) => engine.attributes),
+        count: engines.length,
+      }));
+  }
+
+  public async delete(entityType: EntityType, deleteData: boolean) {
+    const savedObj = await this.engineClient.get(entityType).then(ensureEngineExists(entityType));
+
+    this.options.logger.debug(`Deleting entity store for ${entityType}`);
+
+    await this.options.entityClient.deleteEntityDefinition({ id: savedObj.id, deleteData });
+    await this.engineClient.delete(savedObj.id); // QUESTION: What happens if this fails but the entity definition is successfully deleted?
```

Comment:
Let's discuss this before merging.

What do you think about throwing an error and making sure that the next time the API gets called it doesn't fail when deleting something that is already deleted.

Reply:
I think that makes sense but I'm less clear on whether it's the best option.
I think regardless, we'll have to modify the entity manager code. It either needs to expose a `findEntityDefinition` or not fail on attempting to `delete` a non-existent definition 🤔 

The following flow also feels awkward to me:
1. Delete engine
2. Definition is delete but saved object fails
3. Client gets error describing the current, unsynchronised state
4. Client calls Delete again?

I think 4 should be `FIX` or `RESET` or similar. If we look at this like a finite state machine, the action in 4 should only ever be a possible transition from an error state. 
So maybe we need another route that figures out error recovery?

But this also probably has implications for the UI/UX.

@hop-dev has mentioned the "silent reset" approach, but that was pertaining to calling `INIT` on an existing engine, where it deletes it first and sets it all up again from a fresh, clean state. 
I'm not sure if it fits that nicely on a `DELETE` 🤔

Reply:
Decision was to handle this problem in a later PR 

Tracking this in [#10541](https://github.com/elastic/security-team/issues/10541)